spec/codebreaker_spec.rb

```diff
@@ -0,0 +1,157 @@
+# frozen_string_literal: true
+
+RSpec.describe Game do
+  let(:game) { described_class.new }
+  let(:difficulty) { Codebreaker::Difficulty.new(name: name, attempts: attempts, hints: hints) }
+  let(:difficulty_factory) do
+    Codebreaker::DifficultyFactory.generate Codebreaker::DifficultyFactory::DIFFICULTIES[:easy]
+  end
+  let(:user) { Codebreaker::User.new('Username', difficulty) }
+  let(:name) { 'easy' }
+  let(:attempts) { 15 }
+  let(:hints) { 2 }
+  let(:invalid_name) { 123 }
+
+  describe '#create_user' do
+    context 'success' do
+      it 'creates user' do
+        expect(game.create_user('Username', difficulty)).to be_instance_of(Codebreaker::User)
+        expect(difficulty.attempts).to eq(attempts)
+        expect(difficulty.hints).to eq(hints)
+      end
+    end
+    context 'falsey' do
+      it 'creates user' do
+        expect { game.create_user(invalid_name, difficulty) }.to raise_error(Codebreaker::Errors::ClassValidError)
+        expect { game.create_user(' ', difficulty) }.to raise_error(Codebreaker::Errors::MinLengthError)
+        expect { game.create_user(invalid_name, difficulty) }.to raise_error(Codebreaker::Errors::ClassValidError)
+      end
+    end
+  end
+
+  describe '#create_difficulty' do
+    context 'success' do
+      it 'should have choose right name difficulty' do
+        expect do
+          game.create_difficulty(name).not_to raise_error(Codebreaker::Errors::InvalidDifficultyError)
+        end
+      end
+      it 'should have choose right attempts' do
+        expect(difficulty.attempts).to eq(15)
+      end
+
+      it 'should have choose right hints' do
+        expect(difficulty.hints).to eq(2)
+      end
+      context 'falsey' do
+        it 'enter invalid difficulty' do
+          expect do
+            game.create_difficulty(invalid_name).to raise_error(Codebreaker::Errors::InvalidDifficultyError)
+          end
+        end
+      end
+    end
+
+    describe '#take_attempt' do
+      context 'guess is valid' do
+        it '#take_attempt not change @attempts value if called for the first time' do
+          game.stub(:take_attempt)
```

Comment:
remove it

Reply:
let(:user) { ... }
....
game.instance_variable_set(:@secret_code, 1234)
game.instance_variable_set(:@user, user)
expect { game.take_attempt(1234) }.to eq('++++')